Improved documentation, distance() function for vertices

j-ac opened the pull request:
-Added documentation comments
-vertices now have distance()

j-ac committed:
Improved documentation, distance() function for vertices
-Added documentation comments
-vertices now have distance()

will459 commented on src/lib.rs:
```diff
@@ -136,20 +138,36 @@ pub struct Coordinate3d {
+    ///Coordinate3d::translate for each vertex in the mesh.
     pub fn translate(&mut self, vector: ColumnVector<3>) {
         for vertex in self.vertices.iter_mut() {
             vertex.translate(vector.clone()); //TODO: avoid clone() by implementing + by reference.
```
translate takes by value, you don't need clone here.

will459 commented on src/lib.rs:
```diff
@@ -3,12 +3,13 @@ use std::ops;
 use std::ops::Mul;
 use std::ops::{Index, IndexMut};
 
+///An abstraction over a 2D matrix of f64s. the foundation of this library
```
```suggestion
/// An abstraction over a 2D matrix of f64s. the foundation of this library
```
nit, a space is more readable

will459 commented on src/lib.rs:
```diff
@@ -3,12 +3,13 @@ use std::ops;
 }
 
-//Matrix with one column
+///A Matrix with only one column
```
```suggestion
/// A [`Matrix`] with only one column
```
You can make link references like this.  This works for std:: items like [\`f64\`] above

will459 commented:
You are missing crate level documentation and Coordinate3d.   Adding #![deny(missing_docs)] or #![warn(missing_docs)] to deny or warn about them.

j-ac committed:
Fixed causes of the rejected pull request
-Added crate level documentation
-Fixed missing spaces following comments
-Removed uneccesary clone() call

ben-j-c commented on src/lib.rs:
```diff
@@ -136,20 +138,36 @@ pub struct Coordinate3d {
+            ans += (self.pos[(axis, 1)] + cord.pos[(axis, 1)]).powi(2);
+        }
+
+        ans = ans.sqrt();
+        ans
```
Can't you just `ans.sqrt()` here?

j-ac replied:
Fixed just a second ago in today's commit to PR_1

ben-j-c commented on src/lib.rs:
```diff
@@ -131,28 +138,44 @@ impl<const ROWS: usize, const COLS: usize> ops::Mul<f64> for Matrix<ROWS, COLS>
+            ans += (self.pos[(axis, 1)] + cord.pos[(axis, 1)]).powi(2);
+        }
+
+        ans.sqrt()
```
Misnomer. Should be sum or something like that.  :)

j-ac replied:
Sure. I'll fix

j-ac committed:
Improving a variable name in Corrdinate3d::distance
as per ben-j-c 's suggestion

j-ac committed:
Fixed broken links in documentation

j-ac committed:
more documentation mistakes fixed
